### PR TITLE
o/servicestate/quotas: add functions for getting and setting quotas in state

### PR DESCRIPTION
**Pull request by anonymouse64**
* Add AllQuotas for getting all quotas from the state
* Add GetQuota for getting an individual quota group from the state by name
* Add UpdateQuotas for updating one or many quotas in the state in a single 
  transaction
* Add some units for this, including refactoring to share some of the overlord
  setup with the servicemgr_test.go tests via baseServiceMgrTestSuite

This is orthogonal to https://github.com/snapcore/snapd/pull/10198

### CODE REVIEW

File: overlord/servicestate/quotas.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import (
+	"fmt"
+	"sort"
+
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap/quota"
+)
+
+// AllQuotas returns all currently tracked quota groups in the state. They are
+// validated for consistency before returned and internally updated using
```
Comment (pedronis) [inline overlord/servicestate/quotas.go:31]: being returned ?

File: overlord/servicestate/quotas.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+// ResolveCrossReferences before returned.
```
Comment (pedronis) [inline overlord/servicestate/quotas.go:32]: same?

File: overlord/servicestate/quotas.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	// if the referenced group does not exist we return a nil group
+	return allGrps[name], nil
+}
+
+// UpdateQuotas will update the state quota group map with the provided quota
+// groups. The groups provided will be added on top of the current set of quota
```
Comment (pedronis) [inline overlord/servicestate/quotas.go:65]: maybe for completeness: ... will be replace group states for the same name if present or be added on top of ...

**Review by pedronis — APPROVED** [review]
thanks, couple of small comments

File: overlord/servicestate/quotas.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+// GetQuota returns an individual quota group by name.
+func GetQuota(st *state.State, name string) (*quota.Group, error) {
+	allGrps, err := AllQuotas(st)
+	if err != nil {
+		return nil, err
+	}
+
+	// if the referenced group does not exist we return a nil group
```
Comment (mvo5) [inline overlord/servicestate/quotas.go:59]: Thanks for this coment

**Review by mvo5 — APPROVED** [review]
Thank you

**Pull request merged**